lib/blockchain/chaindb.js

```diff
@@ -1360,6 +1360,10 @@ class ChainDB {
     if (!Array.isArray(addrs))
       addrs = [addrs];
 
+    for (let addr of addrs)
+      if (typeof addr === 'string')
+        addr = Address.fromString(addr);
+
     const coins = [];
 
     for (const addr of addrs) {
```

Comment:
I think Mark is right - You can change THIS for loop to be `let addr of addrs...` and then mutate strings into objects before it starts processing.

Reply:
Thanks, fixed.